cpp/src/arrow/flight/client.h

```diff
@@ -59,7 +59,12 @@ class ARROW_FLIGHT_EXPORT FlightCallOptions {
 
 class ARROW_FLIGHT_EXPORT FlightClientOptions {
  public:
+  /// \brief Root certificates to use for validating server
+  /// certificates.
   std::string tls_root_certs;
+  /// \brief Override the hostname checked by TLS. Insecure, use with
```

Comment:
I don't think we need to mark it insecure?

Reply:
I can shorten the name, but I'd at least like to stress that it's potentially a footgun in production.